### PR TITLE
Refactor Google Cloud Speech Module

### PR DESCRIPTION
**Pull request by IlyaUmanets**
Changes:

* Put empty lines to increase readability
* Add `BILLION` constant
* Use `zero?` method instead of `== 0`
* Don't return `nil` directly. `return` returns `nil` by default.
* Remove unused variable declaration
* Replace `return unless object` with `return if object.nil?`

### CODE REVIEW

**Comment by IlyaUmanets** [comment]
@googlebot, I signed it!

File: google-cloud-speech/lib/google/cloud/speech/version.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@
 module Google
   module Cloud
     module Speech
-      VERSION = "0.27.0"
+      VERSION = "0.27.1"
```
Comment (blowmage) [inline google-cloud-speech/lib/google/cloud/speech/version.rb]: Can you revert the change to the `VERSION`? We change this during the release process, as [documented here](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/RELEASING.md).

**Comment by IlyaUmanets** [comment]
@blowmage, the PR is updated

File: google-cloud-speech/lib/google/cloud/speech/service.rb [hunk 2]
```diff
@@ -29,15 +29,16 @@ class Service
 
         ##
         # Creates a new Service instance.
-        def initialize project, credentials, timeout: nil, client_config: nil
+        def initialize project, credentials, timeout: nil, client_config: {}
```
Comment (blowmage) [inline google-cloud-speech/lib/google/cloud/speech/service.rb]: The problem with this change is what happens if `nil` is passed in. The GRPC code expects a hash, and this change could cause errors. @quartzmo what do you think?

Comment (quartzmo) [reply google-cloud-speech/lib/google/cloud/speech/service.rb]: @blowmage I agree.

@IlyaUmanets What is your motivation for the changes in this PR? They appear to be stylistic refactorings, but I am concerned that despite the project's automated testing, there is a very slight risk of introducing functional changes that I do not see justified by any benefit, unless you can propose one.

**Comment by IlyaUmanets** [comment]
@blowmage, @quartzmo, you are absolutely right, it was a bad change.

File: google-cloud-speech/lib/google/cloud/speech/project.rb [hunk 3]
```diff
@@ -626,7 +626,6 @@ def operation id
         def audio_config encoding: nil, language: nil, sample_rate: nil,
                          max_alternatives: nil, profanity_filter: nil,
                          phrases: nil, words: nil
-          contexts = nil
```
Comment (blowmage) [inline google-cloud-speech/lib/google/cloud/speech/project.rb]: While this doesn’t change functionality, it will issue a warning when `contexts` is used without first being initialized.

**Comment by quartzmo** [comment]
@IlyaUmanets What is your motivation for the changes in this PR? The changes appear to be just stylistic refactorings. I am concerned that despite the project's automated testing, there is a very slight risk of introducing functional changes or bugs. I do not see enough benefit from style changes for this slight risk, personally.

**Comment by quartzmo** [comment]
@IlyaUmanets Thank you for submitting this pull request. For the reason I gave above, we have decided not to accept it at this time. However, we really appreciate the effort: If you have any suggestion for how to functionally improve the google-cloud-ruby packages, please open an issue to discuss, and we will assist you in developing a PR if you want to do so.

**Comment by IlyaUmanets** [comment]
@quartzmo I absolutely agree with you. Thanks a lot !
